pkg/generate/apigwv2_test.go

```diff
@@ -222,11 +222,64 @@ func TestAPIGatewayV2_Route(t *testing.T) {
 	if resp.ApiGatewayManaged != nil {
 		ko.Status.APIGatewayManaged = resp.ApiGatewayManaged
 	}
+	if resp.ApiKeyRequired != nil {
+		ko.Spec.APIKeyRequired = resp.ApiKeyRequired
+	}
+	if resp.AuthorizationScopes != nil {
+		f2 := []*string{}
+		for _, f2iter := range resp.AuthorizationScopes {
+			var f2elem string
+			f2elem = *f2iter
+			f2 = append(f2, &f2elem)
+		}
+		ko.Spec.AuthorizationScopes = f2
+	}
+	if resp.AuthorizationType != nil {
+		ko.Spec.AuthorizationType = resp.AuthorizationType
+	}
+	if resp.AuthorizerId != nil {
+		ko.Spec.AuthorizerID = resp.AuthorizerId
+	}
+	if resp.ModelSelectionExpression != nil {
+		ko.Spec.ModelSelectionExpression = resp.ModelSelectionExpression
+	}
+	if resp.OperationName != nil {
+		ko.Spec.OperationName = resp.OperationName
+	}
+	if resp.RequestModels != nil {
+		f7 := map[string]*string{}
+		for f7key, f7valiter := range resp.RequestModels {
+			var f7val string
+			f7val = *f7valiter
+			f7[f7key] = &f7val
+		}
+		ko.Spec.RequestModels = f7
+	}
+	if resp.RequestParameters != nil {
+		f8 := map[string]*svcapitypes.ParameterConstraints{}
+		for f8key, f8valiter := range resp.RequestParameters {
+			f8val := &svcapitypes.ParameterConstraints{}
+			if f8valiter.Required != nil {
+				f8val.Required = f8valiter.Required
+			}
+			f8[f8key] = f8val
+		}
+		ko.Spec.RequestParameters = f8
+	}
 	if resp.RouteId != nil {
 		ko.Status.RouteID = resp.RouteId
 	}
+	if resp.RouteKey != nil {
+		ko.Spec.RouteKey = resp.RouteKey
+	}
+	if resp.RouteResponseSelectionExpression != nil {
+		ko.Spec.RouteResponseSelectionExpression = resp.RouteResponseSelectionExpression
+	}
+	if resp.Target != nil {
+		ko.Spec.Target = resp.Target
+	}
```

Comment:
OK, so what I'm realizing from the above is that you are setting the `Spec` fields -- presumably to a **default** value that the API Gateway v2 service sets on the `Route` resource? If that is the case, we need to think carefully here about whether we might be inadvertently overwriting field values that were set in the desired `Spec` by the Kubernetes user.

Put another way, if `ko.Spec.Target` was set to some non-nil value already by the Kubernetes user in their CR, then the above Go code will overwrite the `Target` field with whatever was returned in the `CreateRoute` response. There's no way to tell if the values were different.

Potentially we could place some guard expression like this?

```go
        if resp.Target != nil && ko.Spec.Target == nil {
		ko.Spec.Target = resp.Target
	}
```

Reply:
That's fair. I will work on it.
* I am thinking of avoiding writing spec fields for Create
* I will write spec fields for latestKO when we are comparing desiredKO and latestKO objects in case of Update.

Does the rest of PR look okay to you except handling the CreateResponse??